velox/exec/Aggregate.h

```diff
@@ -175,6 +174,13 @@ class Aggregate {
   // 'groups'. No-op for fixed length accumulators.
   virtual void destroy(folly::Range<char**> /*groups*/) {}
 
+  // Returns the type that extractAccumulator will produce. This is
+  // different from resultType_ for complex final aggregations. Needed
+  // for e.g. spilling.
+  virtual TypePtr accumulatorType() const {
```

Comment:
The implementation of this method is incorrect. The comment says "This is different from resultType_", but the implementation is simply returning resultType_. A cleaner change would be to make this method non-virtual, modify Aggregate's constructor to require accumulator type to be specified in addition to result type and return accumulator type here. Otherwise, we'll keep running into spill-specific bugs caused by various aggregate functions not overriding this method.

It would be best to make this change in a separate PR.